Source/Core/VideoCommon/VideoBackendBase.h

```diff
@@ -31,34 +31,34 @@ enum EFBAccessType
 struct SCPFifoStruct
 {
 	// fifo registers
-	volatile u32 CPBase;
-	volatile u32 CPEnd;
+	u32 CPBase;
+	u32 CPEnd;
 	u32 CPHiWatermark;
 	u32 CPLoWatermark;
-	volatile u32 CPReadWriteDistance;
-	volatile u32 CPWritePointer;
-	volatile u32 CPReadPointer;
-	volatile u32 CPBreakpoint;
-	volatile u32 SafeCPReadPointer;
+	u32 CPReadWriteDistance;
+	u32 CPWritePointer;
+	u32 CPReadPointer;
+	u32 CPBreakpoint;
+	u32 SafeCPReadPointer;
 	// Super Monkey Ball Adventure require this.
 	// Because the read&check-PEToken-loop stays in its JITed block I suppose.
 	// So no possiblity to ack the Token irq by the scheduler until some sort of PPC watchdog do its mess.
-	volatile u16 PEToken;
+	u16 PEToken;
 
-	volatile u32 bFF_GPLinkEnable;
-	volatile u32 bFF_GPReadEnable;
-	volatile u32 bFF_BPEnable;
-	volatile u32 bFF_BPInt;
-	volatile u32 bFF_Breakpoint;
+	u32 bFF_GPLinkEnable;
+	u32 bFF_GPReadEnable;
+	u32 bFF_BPEnable;
+	u32 bFF_BPInt;
+	u32 bFF_Breakpoint;
 
-	volatile u32 bFF_LoWatermarkInt;
-	volatile u32 bFF_HiWatermarkInt;
+	u32 bFF_LoWatermarkInt;
+	u32 bFF_HiWatermarkInt;
 
-	volatile u32 bFF_LoWatermark;
-	volatile u32 bFF_HiWatermark;
+	u32 bFF_LoWatermark;
+	u32 bFF_HiWatermark;
 
 	// for GP watchdog hack
-	volatile u32 isGpuReadingData;
+	u32 isGpuReadingData;
```

Comment:
I'm pretty sure isGpuReadingData, at the very least, needs to be volatile/atomic.